### ISSUE TITLE
Base.copy! raises METHODERROR from transfer! in CUDAdrv

### ISSUE THREAD

**Issue by jamesjscully** [JuliaGPU/CuArrays.jl#80]
It boils down to this:
```
a =  cu(zeros(100))
b = cu(fill(1,100))
```
then 
`copy!(a,b)`
throws
`ERROR: MethodError: no method matching transfer!(::CUDAdrv.Mem.Buffer, ::CUDAdrv.Mem.Buffer)`

From CuArrays the method for copy calls Mem.transfer from CUDAdrv's memory.jl module:
```
function Base.copy!(dst::CuArray{T}, src::CuArray{T}) where T
    @assert length(dst) == length(src)
    Mem.transfer!(unsafe_buffer(dst), unsafe_buffer(src))
    return dst
end
```
But Mem.transfer! requires at least the number of bytes:
```
function transfer!(dst::Buffer, src::Buffer, nbytes::Integer,
                   stream::CuStream=CuDefaultStream(); async::Bool=false)
    if async
        @apicall(:cuMemcpyDtoDAsync,
                 (Ptr{Cvoid}, Ptr{Cvoid}, Csize_t, CuStream_t),
                 dst, src, nbytes, stream)
    else
        @assert stream==CuDefaultStream()
        @apicall(:cuMemcpyDtoD,
                 (Ptr{Cvoid}, Ptr{Cvoid}, Csize_t),
                 dst, src, nbytes)
    end
end
```
It also seems like this should be caught in Pkg.test("CuArrays").

**Comment by jamesjscully** [comment]
Simply changing 
`Mem.transfer!(unsafe_buffer(dst), unsafe_buffer(src))`
to 
`Mem.transfer!(unsafe_buffer(dst), unsafe_buffer(src), sizeof(source))`
solves the problem.

**Comment by maleadt** [comment]
Looks like this was fixed in 5a11137a68e2cb2a7e8f1348280b089a3555b569... So +1 on #82 :slightly_smiling_face:
